edx_management_commands/management_commands/management/commands/manage_group.py

```diff
@@ -0,0 +1,131 @@
+"""
+Management command `manage_group` is used to idempotently create Django groups
+and set their permissions by name.
+"""
+
+from django.apps import apps
+from django.contrib.auth.models import Group, Permission
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.utils.translation import gettext as _
+
+
+class Command(BaseCommand):
+    # pylint: disable=missing-docstring
+
+    help = 'Creates the specified group, if it does not exist, and sets its permissions.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('group_name')
+        parser.add_argument('--remove', dest='is_remove', action='store_true')
+        parser.add_argument('-p', '--permissions', nargs='*', default=[])
+
+    def _handle_remove(self, group_name):
+
+        try:
+            Group.objects.get(name=group_name).delete()  # pylint: disable=no-member
+            self.stderr.write(_('Removed group: "{}"').format(group_name))
+        except Group.DoesNotExist:
+            self.stderr.write(_('Did not find a group with name "{}" - skipping.').format(group_name))
+            return
+
+    @transaction.atomic
+    def handle(self, group_name, is_remove, permissions=None, *args, **options):
+
+        if is_remove:
+            return self._handle_remove(group_name)
+
+        old_permissions, new_permissions = [], []
+        group, created = Group.objects.get_or_create(name=group_name)  # pylint: disable=no-member
+
+        if created:
```

Comment:
Is there a reason that remove is handled by a separate method, but the other logic isn't split out? Might be nice to break up this 100-line-ish method.